### ISSUE TITLE
[bug]: Can't Log in to new Desktop App on Linux

### ISSUE THREAD

**Issue by carsonbird** [hoppscotch/hoppscotch#3721]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behavior

When I try to log in to the desktop app using my email and a magic link on my Linux machine, I can't.

**Details:**

- Ubuntu 23.10 with KDE Plasma.
- Using the Hoppscotch Desktop App (the thing that's in alpha)
- Attempt to log into my hoppscotch.io workspace through the desktop app, using a magic link and my email address. No Google sign-in or anything, just a direct email address.
- I am receiving the magic link emails, but, number 1, clicking them doesn't link into the desktop app, it opens in the browser; and number 2, the browser says "your're logged in, you can go back to the app now" - but the app hasn't logged in.
- Inspecting the desktop app console, I see the following errors. These appear when the app is initially loaded, long before I attempt to sign in. Not sure if they're related.
- Attempting to save the magic link as a desktop shortcut, then force it to be opened with Hoppscotch also does nothing.

```
[Error] TypeError: undefined is not an object (evaluating 'gapi.iframes.getContext')
	callback (index-95f3a832.js:3949:360)
	(anonymous function) (api.js:29)
	Ja (api.js:21:705)
	E (api.js:28:502)
	(anonymous function) (api.js:29:83)
	(anonymous function) (api.js:29:146)
	Global Code (cb=gapi.loaded_0:1)
```

```
[Error] Error: B`tauri
	(anonymous function) (api.js:29:638)
	(anonymous function) (api.js:29)
	Ja (api.js:21:705)
	E (api.js:28:502)
	(anonymous function) (api.js:29:83)
	(anonymous function) (api.js:29:146)
	Global Code (cb=gapi.loaded_1:1)
```

```
[Error] Cross-origin redirection to http://developers.google.com/ denied by Cross-Origin Resource Sharing policy: Origin tauri://localhost is not allowed by Access-Control-Allow-Origin. Status code: 301
```

```
[Error] XMLHttpRequest cannot load https://apis.google.com/_/jserror?<<REDACTED>> due to access control checks.
```

```
[Error] Failed to load resource: Cross-origin redirection to http://developers.google.com/ denied by Cross-Origin Resource Sharing policy: Origin tauri://localhost is not allowed by Access-Control-Allow-Origin. Status code: 301 (jserror, line 0)
```

So the issues seem to be that either the sign-in is being blocked by CORS, the magic-link webpage isn't able to talk to the desktop app and give it the authenticated stuff, or that the desktop app isn't receiving deep links from the magic link in the email.

### Steps to reproduce

See the above.

### Environment

Production

### Version

Cloud

**Comment by carsonbird** [comment]
Some additional information - I pasted the link into a private session in a new browser, so there wouldn't be any cookie/caching issues. The browser-portion of the authentication seemed to work fine - and then upon returning the desktop app and inspecting the network calls, I can see that the app did successfully receive the login/authentication information from somewhere, because (among other things) it proceeds to make a POST request to https://identitytoolkit.googleapis.com/v1/accounts:lookup?key=XXX and it does include information it could have only received from the authentication flow (like my name). See below for more. However, the app doesn't finish the login process, and just sits there on the "magic link" popup. Clicking "dismiss" dismisses the dialog, but doesn't finish the login.

```json
POST https://identitytoolkit.googleapis.com/v1/accounts:lookup?key=XXX

{
  "kind": "identitytoolkit#GetAccountInfoResponse",
  "users": [
    {
      "localId": "XXX",
      "email": "XXX",
      "displayName": "XXX",
      "emailVerified": true,
      "providerUserInfo": [
        {
          "providerId": "password",
          "displayName": "XXX",
          "federatedId": "XXX",
          "email": "XXX",
          "rawId": "XXX"
        }
      ],
      "validSince": "1703634711",
      "lastLoginAt": "XXX",
      "createdAt": "XXX",
      "emailLinkSignin": true,
      "lastRefreshAt": "2024-01-04T22:55:19.150Z"
    }
  ]
}
```

**Comment by luancgs** [comment]
Also unable to login on linux desktop app. Using Linux Mint 21.1 with Cinnamon and trying to login using google SSO, the app opens the google page on my browser, but after returning, it just loads forever.

**Comment by adaniloff** [comment]
Same here, unable to login on my PopOs (using Pop!_OS 22.04 LTS). Infinite loading when I try to use google or email auth.

**Comment by rochacbruno** [comment]
same here, just tried to login from arch linux, doesnt work with any of the login methods.

v2023.12.6

**Comment by HeIIow2** [comment]
I got the same issue.

**Comment by bfsgr** [comment]
Probably related to https://github.com/tauri-apps/tauri/issues/2327

**Comment by noomly** [comment]
I'm also affected by this issue on Archlinux. Still no workaround?

**Comment by baaslaawe** [comment]
linux Desktop not work login we have try many times for linux like ubuntu Desktop not close login it loading

**Comment by AlphaNecron** [comment]
Bump, having been struggling with this bug for a while on Arch Linux, I guess the only workaround here is to patch `webkit2gtk` to disable `web-security` but that also means lowered security.

**Comment by AlphaNecron** [comment]
Technically, I can just use Hoppscotch anonymously but obviously no syncing lol.

**Comment by mirzahilmi** [comment]
same issue, im using NixOS 24.05

**Comment by 0xi4o** [comment]
I was able to get login working by copying the link, pasting it in Firefox, and logging in from there. Then it was able to connect to the desktop app and log me in. Didn't work in other browsers so it has to be Firefox. YMMV.

**Comment by rochacbruno** [comment]
I confirm ^, after downloading an upgraded version of the AppImage it can login now if I do using Firefox.

**Comment by nicolasCinzer** [comment]
I agree with @rochacbruno, if a try to login using Firefox as browser, the login is successfull

**Comment by kayprogrammer** [comment]
> I agree with @rochacbruno, if a try to login using Firefox as browser, the login is successfull

Worked for me. Thank you very much

**Comment by carsonbird** [comment]
Just downloaded the latest appimage, attempted to use GitHub login through Firefox, doesn't work. Gets pretty close, the Hoppscotch app can see that the login flow occurred in Firefox, so they're communicating. But the Hoppscotch app just sits and spins in the "Login to Hoppscotch" modal after that.

**Comment by carsonbird** [comment]
The app also says that it opens my browser (Brave), but it doesn't. Clicking the "Open Again" button doesn't do anything either.

**Comment by carsonbird** [comment]
Network shows that the authentication information is received by the app. So the issue is probably in the desktop app itself, and how it handles the received data.

**Comment by liyasthomas** [comment]
Thanks for your patience and valuable feedback. We’ve fixed this bug in the latest release.
[Download the latest builds](https://hoppscotch.com/download) and provide your valuable feedback.

Closing this ticket as this bug is now fixed in the recent version. Please feel free to reach out if you have any other concerns.
